improve docs (#39)
carvel-dev/imgpkg: docs/README.md

# Documentation

## What is imgpkg

> `imgpkg` is a tool that allows users to store a set of arbitrary files as an OCI image. One of the driving use cases is to store Kubernetes configuration (plain YAML, ytt templates, Helm templates, etc.) in OCI registry as an image.

Reviewer: Are there other use cases we want to highlight? If not, I'd prefer "The driving use case..."
```suggestion
`imgpkg` is a tool that allows users to store a set of arbitrary files as an OCI image. The driving use case is saving and sharing Kubernetes configuration files in various formats: plain YAML, ytt templates, Helm templates, etc.
```

Author: "in OCI registry as an image." this bit is important.

> Primary concept imgpkg introduces is a [bundle](resources.md#Bundle) which is an OCI image that holds 0+ arbitrary files and 0+ references to dependent OCI images. This allows imgpkg to copy bundles and their dependent images across registries (online and offline).

Reviewer: ```suggestion
`imgpkg`'s primary concept is a [bundle](resources.md#Bundle), which is an OCI image that holds 0+ arbitrary files and 0+ references to dependent OCI images. With this concept, `imgpkg` is able to copy bundles and their dependent images across registries (both online and offline).
```

> ![Bundle diagram](images/bundle-diagram.png)

Reviewer: This looks awesome, great diagram!

## Example Workflows

> - [Basic workflow](basic-workflow.md) shows how to create, push and pull bundles by taking simple Kubernetes application

Reviewer: ```suggestion
- [Basic Workflow](basic-workflow.md) shows how to create, push, and pull bundles with a simple Kubernetes application
```

> - [Air-gapped workflow](air-gapped-workflow.md) shows how to copy bundles from one registry to another to enable running Kubernetes application without relience on external registries

Reviewer: ```suggestion
- [Air-gapped Workflow](air-gapped-workflow.md) shows how to copy bundles from one registry to another, to enable running Kubernetes applications without relying on external (public) registries
```

### Reference

- [Authentication to registry](auth.md)
- [Resources](resources.md) describes concepts and data formats

## Commands

`imgpkg` supports four commands:

> - [`push`](commands.md#push) a bundle from local directory to a registry. 

Reviewer: ```suggestion
- [`push`](commands.md#push) a bundle from a local directory to a registry. 
```
